guillotina/asgi.py

```diff
@@ -112,6 +112,7 @@ def build_middleware_stack(self, settings):
 
         if not resp.prepared:
             await resp.prepare(request)
+            await resp.send_body()
```

Comment:
Shouldn't it check if body has started on a different if, may it be prepared and not sent?

Reply:
My idea was: if you call `prepare()` in the service (like in the download endpoints), you know what you are doing and you are responsible of managing the response flow. The `send_body()` sends the content response.body to the asgi server, but the response.body could be empty in case of streaming responses.

Reply:
I think the best solution/fix is adding:

```python
if not resp.eof:
    await resp.send_eof()
```

(this ensures the asgi server doesn't wait for the response data indefinitely)

@bloodbare agree?

Reply:
agree, websockets will not reach this point until they are done, isn't it ?

Reply:
Exactly